Add concept of parent and sub-topics, where sub-topics are added automatically when

Will-Howard opened the pull request:
Add the concept of parent and sub tags, where parent tags are applied automatically when a sub tag is applied. The initial use case is to make "AI Safety" a parent topic to "AI Alignment", "Artificial Intelligence"

The logic for when to apply a parent tag is:
 - when a subtag is applied to a post for the first time (i.e. there were no previous votes on it's releveance), apply the same vote to the parent tag
 - otherwise do nothing

## UI changes:
### Show parent and subtopics in the popout preview (note: the "+2 sub-topics" isn't there anymore)
![](https://user-images.githubusercontent.com/77623106/183945341-75c193f2-f6c5-4c16-8dd8-a5849ab73a04.png)
![](https://user-images.githubusercontent.com/77623106/183945571-95a7609c-e9fe-4ce1-8137-f08e83a2648a.png)

### Link to parent and subtopics in tag pages
![](https://user-images.githubusercontent.com/77623106/183945701-bd768b0c-92cd-4bf2-87a5-ab0ee2c2f1b9.png)
![](https://user-images.githubusercontent.com/77623106/183945749-4b1ff6af-293e-4384-8c6c-7a6124e3d94e.png)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202780727457004) by [Unito](https://www.unito.io)

Will-Howard commented on packages/lesswrong/lib/collections/tags/custom_fields.ts:
```diff
@@ -0,0 +1,63 @@
+    control: 'TagSelect',
+    onInsert: async (tag) => {
+      if (tag.parentTagId) {
+        // don't allow chained parent tag relationships
```
This is to avoid having to detect dependency cycles

Xodarap commented:
I don't see the "+2 subtopics" UI bit. Did that change?

![image](https://user-images.githubusercontent.com/221511/185510112-5a2db438-e236-4721-b9c8-a441a29c202f.png)

Xodarap reviewed (approved):
One minor comment, but otherwise this looks good to me.

Will-Howard commented:
> EDIT: yes, you said that changed. But I kind of liked it – is there a reason you deleted it?

Ruby said he thought it was unnecessary and I didn't have a strong preference so I removed it. I can add it back in in another PR if you are strongly pro. If I added it back in it would now say "+ Parent Topic" because in the first version I was adding subtopics when a parent topic was added, now it's the other way round
